### PR TITLE
Add tests for when the RLP payload is <= 32 bytes

### PR DESCRIPTION
**Pull request by gballet**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1453,4 +1453,52 @@ mod tests {
             .map(|i| u8::from_str_radix(&s[i..i + 2], 16).unwrap())
             .collect()
     }
+
+    #[test]
+    fn check_payload_length_exactly_32_bytes() {
```
Comment (s1na) [inline src/lib.rs:1581]: Just to make sure, you omitted leaf with length < 32 bytes on purpose?

Comment (gballet) [reply src/lib.rs:1581]: Nope,  but I had a merge mishap so I assume it got lost during that time. I have also lost a fix for the empty leaf issue :sweat: 

Anyhow, I'll rewrite it as it shouldn't be too long.

File: src/lib.rs [hunk 2]
```diff
@@ -1444,4 +1444,63 @@ mod tests {
...
+    }
+
+    #[test]
+    fn check_leaf_length_less_than_32_bytes() {
+        let mut root = Node::default();
+
+        root = insert_leaf(&mut root, &NibbleKey::from(vec![1u8; 2]), vec![1u8; 20]).unwrap();
+        assert_eq!(
+            root.hash(),
+            vec![216, 130, 32, 17, 148, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
```
Comment (s1na) [inline src/lib.rs:1602]: Ah this is the edge case I was mentioning. Here I think the root hash should be a hash of this leaf

**Review by s1na — APPROVED** [review]
Looks good! I wasn't quite sure about the name `composition` but I had no better alternative honestly, so :+1: :smile:

**Pull request merged**